### PR TITLE
@alloy =>Fixes to fair search when there is no local show cache

### PR DESCRIPTION
**Pull request by orta**
It uses the API instead.

Full context [in Slack](https://artsy.slack.com/archives/help/p1472756429000061) - but roughly we were assuming there was always a local cache of shows on the fair object, but if we don't have a map it's not downloaded, so the search would not show any partners - now it checks for the existence of the cache before relying on it.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |263.7ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Auction/AuctionViewController.swift#L199) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
205.6ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-fair_search/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
173.4ms | [WebSocket.swift](/artsy/eigen/tree/orta-fair_search/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
80.3ms | [Observable.swift](/artsy/eigen/tree/orta-fair_search/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
77.0ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
74.7ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
72.4ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L49) | @objc override func viewDidLoad()
65.4ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
63.3ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)
62.2ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/orta-fair_search/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Gemfile.lock [hunk 1]
```diff
@@ -77,38 +77,50 @@ GEM
...
     escape (0.0.4)
     faraday (0.9.2)
       multipart-post (>= 1.2, < 3)
-    faraday-http-cache (1.3.0)
+    faraday-http-cache (1.3.1)
       faraday (~> 0.8)
     fourflusher (0.3.0)
     fuzzy_match (2.0.4)
     git (1.3.0)
+    gitlab (3.7.0)
```
Comment (alloy) [inline Gemfile.lock]: Are we moving? o_O

Comment (orta) [inline Gemfile.lock]: :P - afraid the gitlab support in Danger is complex enough to warrant the gem

Comment (alloy) [inline Gemfile.lock]: But, why is it not optional? E.g. `danger-gitlab`

Comment (orta) [inline Gemfile.lock]: Aye, that makes sense, I should do this ( I do it for the JS one for example )

Comment (alloy) [inline Gemfile.lock]: 👍

Comment (orta) [inline Gemfile.lock]: it's not happening today though, as that's a server version change, for now `:dealwithit:`

**Comment by sarahscott** [comment]
ace

**Pull request merged**